plugins/catalyst/python/pyproject.toml

```diff
@@ -0,0 +1,298 @@
+manylinux-ppc64le-image = "manylinux_2_28"
+manylinux-s390x-image = "manylinux_2_28"
+test-skip = [
+  "cp3*t-*", # no freethreading qiskit wheels
+  "cp*-win_arm64", # no numpy, qiskit, ... wheels
+]
+
+[tool.cibuildwheel.linux]
+environment = { DEPLOY = "ON" }
+
+[tool.cibuildwheel.macos]
+environment = { MACOSX_DEPLOYMENT_TARGET = "10.15" }
+
+[[tool.cibuildwheel.overrides]]
+select = "*-macosx_arm64"
+environment = { MACOSX_DEPLOYMENT_TARGET = "11.0" }
+
+[tool.uv]
+required-version = ">=0.5.20"
+reinstall-package = ["mqt-catalyst-plugin"]
+
+[tool.uv.sources]
+mqt-catalyst-plugin = { workspace = true }
+
+[dependency-groups]
+build = [
+  "pybind11>=2.13.6",
+  "scikit-build-core>=0.11.1",
+  "setuptools-scm>=8.2",
+  "mqt-core>=3.0.0",
+  "pennylane @ https://test-files.pythonhosted.org/packages/a4/ae/b8b0fdcd91fe0e87d699d46938cfee0531e8093042dd9eff11abc247c02f/pennylane-0.42.0.dev27-py3-none-any.whl",
+  "pennylane-lightning @ https://test-files.pythonhosted.org/packages/4d/99/1cab1348e3b839f0c8df515dc6b2d347c5e913361ff65e6fa1f197c0a67a/pennylane_lightning-0.42.0.dev16-cp312-cp312-macosx_13_0_arm64.whl",
+  "pennylane-lightning-kokkos @ https://test-files.pythonhosted.org/packages/25/97/9707f73d2ed6e2549e4e099d0083481c144e984227e147b44cd2838535e6/pennylane_lightning_kokkos-0.42.0.dev16-cp312-cp312-macosx_13_0_arm64.whl",
+  "pennylane-catalyst @ https://test-files.pythonhosted.org/packages/00/1f/791910226d7363ba8272add71529c637426d1ac225a4e582ef8612b7ecee/pennylane_catalyst-0.12.0.dev37-cp312-abi3-macosx_13_0_arm64.whl",
+]
+docs = [
+  "furo>=2024.8.6",
+  "myst-nb>=1.2.0",
+  "setuptools-scm>=8.2",
+  "sphinx-autoapi>=3.6.0",
+  "sphinx-copybutton>=0.5.2",
+  "sphinx-design>=0.6.1",
+  "sphinxcontrib-bibtex>=2.6.3",
+  "sphinxcontrib-svg2pdfconverter>=1.3.0",
+  "sphinxext-opengraph>=0.10.0",
+  "openqasm-pygments>=0.1.2",
+  "breathe>=4.36.0",
+  "graphviz>=0.20.3",
+  "sphinx>=7.4.7",
+  "sphinx>=8.1.3; python_version >= '3.10'",
+  "sphinx>=8.2.3; python_version >= '3.11'",
+]
+test = [
+  "pytest>=8.3.5",
+  "pytest-console-scripts>=1.4.1",
+  "pytest-cov>=6.1.1",
+  "numpy>=2.1; python_version >= '3.13'",
+  "mqt-core>=3.0.0",
+  "pennylane @ https://test-files.pythonhosted.org/packages/a4/ae/b8b0fdcd91fe0e87d699d46938cfee0531e8093042dd9eff11abc247c02f/pennylane-0.42.0.dev27-py3-none-any.whl",
+  "pennylane-lightning @ https://test-files.pythonhosted.org/packages/4d/99/1cab1348e3b839f0c8df515dc6b2d347c5e913361ff65e6fa1f197c0a67a/pennylane_lightning-0.42.0.dev16-cp312-cp312-macosx_13_0_arm64.whl",
+  "pennylane-lightning-kokkos @ https://test-files.pythonhosted.org/packages/25/97/9707f73d2ed6e2549e4e099d0083481c144e984227e147b44cd2838535e6/pennylane_lightning_kokkos-0.42.0.dev16-cp312-cp312-macosx_13_0_arm64.whl",
+  "pennylane-catalyst @ https://test-files.pythonhosted.org/packages/00/1f/791910226d7363ba8272add71529c637426d1ac225a4e582ef8612b7ecee/pennylane_catalyst-0.12.0.dev37-cp312-abi3-macosx_13_0_arm64.whl",
```

Comment:
These are not test dependencies.
They are regular dependencies of the plugin. and should be marked as such.
Someone installing this package will have to have mqt-core installed and will have to have catalyst installed.